src/client/clientmap.cpp

```diff
@@ -445,12 +445,14 @@ void ClientMap::renderMap(video::IVideoDriver* driver, s32 pass)
 
 	// Render all mesh buffers in order
 	drawcall_count += draw_order.size();
+	u16 i = 0;
 
 	for (auto &descriptor : draw_order) {
 		scene::IMeshBuffer *buf = descriptor.getBuffer();
 
 		// Check and abort if the machine is swapping a lot
-		if (draw.getTimerTime() > 2000) {
+		// Avoid checking too frequently because getTimerTime() is expensive
+		if (++i % 128 == 0 && draw.getTimerTime() > 2000) {
```

Comment:
I think we can just drop this code, aborting rendering of the world halfway if it takes too long is not good behaviour

Reply:
No regrets :)